support-workers/src/typescript/emailFields/paymentDescription.ts

```diff
@@ -0,0 +1,194 @@
+import { partition } from '@modules/arrayFunctions';
+import type { IsoCurrency } from '@modules/internationalisation/currency';
+import { getCurrencyInfo } from '@modules/internationalisation/currency';
+import { isNonEmpty } from '@modules/nullAndUndefined';
+import { BillingPeriod } from '@modules/product/billingPeriod';
+import dayjs from 'dayjs';
+import type { Payment, PaymentSchedule } from '../model/paymentSchedule';
+
+function billingPeriodNoun(period: BillingPeriod): string {
+	switch (period) {
+		case BillingPeriod.Monthly:
+			return 'month';
+		case BillingPeriod.Quarterly:
+			return 'quarter';
+		case BillingPeriod.Annual:
+			return 'year';
+		default:
+			return period;
+	}
+}
+
+export function formatPrice(amount: number): string {
+	return amount.toFixed(2);
+}
+
+export function priceWithCurrency(
+	currency: IsoCurrency,
+	amount: number,
+): string {
+	return `${getCurrencyInfo(currency).glyph}${formatPrice(amount)}`;
+}
+
+export function firstPayment(paymentSchedule: PaymentSchedule): Payment {
+	if (
+		paymentSchedule.payments.length === 0 ||
+		paymentSchedule.payments[0] === undefined
+	) {
+		throw new Error('Payment schedule has no payments');
+	}
+	return paymentSchedule.payments.reduce<Payment>(
```

Comment:
Are you able to use `earlierPayment` here? Does TS have enough to know that the array is non-empty at this point?